integreat_cms/cms/views/media/media_actions.py

```diff
@@ -81,12 +81,15 @@ def get_directory_content_ajax(request, region_slug=None):
     directory = None
     if request.GET.get("directory"):
         directory = get_object_or_404(
-            Directory.objects.filter(Q(region=region) | Q(region__isnull=True)),
+            Directory.objects.filter(
+                Q(region=region, is_hidden=False) | Q(region__isnull=True),
+            ),
             id=request.GET.get("directory"),
         )
 
     media_files = MediaFile.objects.filter(
-        Q(region=region) | Q(region__isnull=True), Q(parent_directory=directory)
+        Q(region=region, is_hidden=False) | Q(region__isnull=True),
+        Q(parent_directory=directory),
     )
     directories = Directory.objects.filter(
         Q(region=region) | Q(region__isnull=True), parent=directory
```

Comment:
```suggestion
        Q(region=region) | Q(region__isnull=True, is_hidden=False), parent=directory
```
The directories need to be filtered as well.